sprint_6-solution

Altron2108 opened the pull request.

Altron2108 committed:
Все еще пытаюсь зафорсить во вторую ветку

Altron2108 committed:
Все еще пытаюсь зафорсить во вторую ветку

Altron2108 committed:
Все еще пытаюсь зафорсить во вторую ветку

johnskyi commented on test/InMemoryHistoryManagerTest.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    @Test
+    void testHistoryPreservesOrder() {
```
Есть некоторое соглашение по именованию тестовых методов. их принято именовать по схеме какойМетодТестируемОжидаемыйРезультат. Переименуй пожалуйста сразу везде

Altron2108 committed:
epicConstructor_ValidInput_EpicCreatedSuccessfully:

epicConstructor — тестируем конструктор класса Epic.
ValidInput — ввод корректных данных (название и описание эпика).
EpicCreatedSuccessfully — ожидаемый результат: эпик создается корректно, и поля заполняются верными значениями.

addSubtask_SubtaskIdAdded_SubtaskAddedToEpic:

addSubtask — тестируем метод addSubtask.
SubtaskIdAdded — добавляем ID подзадачи в эпик.
SubtaskAddedToEpic — ожидаемый результат: подзадача должна быть успешно добавлена в список подзадач эпика.

Altron2108 committed:
addTaskToHistory_TaskAdded_HistoryContainsOneTask:

addTaskToHistory — тестируем метод добавления задачи в историю.
TaskAdded — добавляем задачу в историю.
HistoryContainsOneTask — ожидаемый результат: в истории должна содержаться одна задача.

Altron2108 committed:
addTasksToHistory_HistoryPreservesCorrectOrder — тестирует, что история сохраняет порядок добавления задач.
addSameTaskToHistory_TaskMovesToEnd — тестирует, что повторное добавление задачи перемещает её в конец списка.
addDuplicateTask_HistoryDoesNotContainDuplicates — проверяет, что история не содержит дубликатов задач.
getHistory_HistoryReturnsUnmodifiableList — проверяет, что метод getHistory возвращает немодифицируемый список.

Altron2108 committed:
createTask_TaskIsCreatedSuccessfully — проверяет, что задача создается успешно.
getTaskById_TaskIsRetrievedSuccessfully — проверяет успешное получение задачи по идентификатору.
updateTask_TaskIsUpdatedSuccessfully — проверяет успешное обновление задачи.
deleteTaskById_TaskIsDeletedSuccessfully — проверяет успешное удаление задачи.
createSubtask_SubtaskIsCreatedSuccessfully — проверяет успешное создание подзадачи.
updateSubtask_SubtaskIsUpdatedSuccessfully — проверяет успешное обновление подзадачи.
deleteSubtaskById_SubtaskIsDeletedSuccessfully — проверяет успешное удаление подзадачи.
epicStatusUpdate_EpicStatusUpdatesCorrectly — проверяет корректное обновление статуса эпика.

Altron2108 committed:
!!!

Altron2108 committed:
getDefaultTaskManager_TaskManagerIsInitialized — проверяет, что метод getDefaultTaskManager возвращает инициализированный экземпляр менеджера задач.
getDefaultHistoryManager_HistoryManagerIsInitialized — проверяет, что метод getDefaultHistoryManager возвращает инициализированный экземпляр менеджера истории.

Altron2108 committed:
constructor_ShouldInitializeFieldsCorrectly — метод описывает, что тестируется (конструктор) и ожидаемый результат (поля должны быть корректно инициализированы).

Altron2108 committed:
getTitle_ShouldReturnInitialTitle — метод проверяет, что заголовок задачи остается первоначальным после создания.
getDescription_ShouldReturnInitialDescription — метод проверяет, что описание задачи остается первоначальным после создания.
getStatus_ShouldReturnInitialStatus — метод проверяет, что статус задачи остается первоначальным после создания.

Altron2108 committed:
createTask_ShouldAssignUniqueIds — имя метода отражает тестируемую функцию (createTask) и ожидаемый результат (ShouldAssignUniqueIds). Тест проверяет, что задачи, созданные менеджером, имеют уникальные идентификаторы.

Altron2108 committed:
taskConstructor_ShouldInitializeFieldsCorrectly — название указывает, что тестируется конструктор задачи и проверяется корректная инициализация полей.
setTitle_setDescription_setStatus_ShouldUpdateTaskFields — название указывает, что тестируются методы setTitle, setDescription и setStatus, и они должны корректно обновлять соответствующие поля задачи.
